coffeecard/CoffeeCard.Library/Services/AccountService.cs

```diff
@@ -289,20 +289,25 @@ private async Task AnonymizeUser(User user)
             user.DateUpdated = DateTime.Now;
             user.PrivacyActivated = true;
             user.UserState = UserState.Deleted;
-            await _context.SaveChangesAsync();            
+            await _context.SaveChangesAsync();
         }
-        
-        private User VerifyTokenClaimAndUser(string token)
+
+        private async Task<User> VerifyTokenClaimAndUserAsync(string token)
         {
+            if (await _tokenService.ValidateTokenAsync(token))
```

Comment:
Should it not be the negated case?

Reply:
I guess you could have a test case for this one 😉

Reply:
Added some tests, turns out one of the other cases was also wrong